compiler/ilgen/TypeDictionary.cpp

```diff
@@ -577,6 +586,47 @@ TypeDictionary::CloseStruct(const char *structName)
    }
 
 TR::IlType *
+TypeDictionary::DefineUnion(const char *unionName)
+   {
+   TR_HashId unionID=0;
+   _unionsByName->locate(unionName, unionID);
+
+   UnionType *newType = new (PERSISTENT_NEW) UnionType(unionName, _trMemory);
+   _unionsByName->add(unionName, unionID, (void *) newType);
+
+   return (TR::IlType *) newType;
+   }
+
+void
+TypeDictionary::UnionField(const char *unionName, const char *fieldName, TR::IlType *type)
+   {
+   TR_HashId unionID=0;
+   if (_unionsByName->locate(unionName, unionID))
+      {
+      UnionType *unionType = (UnionType *) _unionsByName->getData(unionID);
+      unionType->AddField(fieldName, type);
```

Comment:
Should we check if a field by the same name was added to this union before ? There seems to be no protection either here or in AddField in case some code ended up calling these apis twice for the same field in a union. I know this is not a good practice but it is cleaner to think duplication won't happen.